File: java/FactoryMethod/src/factory/PostCodeFactory.java
package factory;

public class PostCodeFactory {
	enum Country {
		US, UK, IN;
	}
	
	public static PostCode getInstance(Country country, String postalCode)
	{
		if (country == Country.US)
			return new _______(postalCode); //> 1
		if (country == Country.UK)
			return new ______(postalCode); //> 2
		if (country == Country.IN)
			return new ______(postalCode); //> 3
		
		return null;
	}
}

File: ruby/FactoryMethod/post_code_factory.rb
require_relative 'us_post_code'
require_relative 'uk_post_code'
require_relative 'in_post_code'

class PostCodeFactory
  def self.get_instance(country, postcode)
    if country == "US"
      return _____.new(postcode) # > 1
    elsif country == "UK"
      return _____.new(postcode) # > 2
    elsif country == "IN"
      return _____.new(postcode) # > 3
    end

    raise 'Country is not supported'
  end
end